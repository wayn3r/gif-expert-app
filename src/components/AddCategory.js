import { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ addCategory }) => {
    const [inputValue, setInputValue] = useState('')
    const handleInputChange = e => {
        setInputValue(e.target.value)
    }

    const handleSubmit = e => {
        e.preventDefault()
        if (inputValue.trim().length > 2) addCategory(inputValue)
        setInputValue('')
    }
    return (
        <form onSubmit={handleSubmit} className='add-category__form'>
            <input
                type="text"
                value={inputValue}
                placeholder="Add a category"
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired,
}

export default AddCategory
