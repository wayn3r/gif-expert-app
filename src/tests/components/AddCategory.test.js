import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe('Pruebas en el archivo AddCategory ', () => {
    const addCategory = jest.fn()
    const preventDefault = () => {}
    let wrapper = shallow(<AddCategory addCategory={addCategory} />)

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory addCategory={addCategory} />)
    })

    test('debe mostrar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe de cambiar el input text', () => {
        const value = 'Hola mundo'
        const input = wrapper.find('input')
        input.simulate('change', { target: { value } })
        // console.log(input.prop('value'))
        // expect(input.prop('value')).toBe(value)
    })
    test('no debe de postear la informacion con submit', () => {
        const form = wrapper.find('form')
        form.simulate('submit', { preventDefault })

        expect(addCategory).not.toHaveBeenCalled()
    })

    test('debe de llamar el addCategory y limpiar la caja de texto', () => {
        // simular el inputChange
        const value = 'Hola mundo'
        const input = wrapper.find('input')
        input.simulate('change', { target: { value } })
        // simular el submit
        const form = wrapper.find('form')
        form.simulate('submit', { preventDefault })
        // addCategory debe de ser llamado
        expect(addCategory).toHaveBeenCalledWith(value)
        // input debe de estar en blanco
        expect(input.prop('value')).toBe('')
    })
})
