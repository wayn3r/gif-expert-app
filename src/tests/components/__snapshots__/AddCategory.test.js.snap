// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo AddCategory  debe mostrar el componente correctamente  1`] = `
<form
  className="add-category__form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add a category"
    type="text"
    value=""
  />
</form>
`;
